qemu.go

```diff
@@ -542,16 +525,23 @@ func (q *qemu) init(pod *Pod) error {
 	q.config = pod.config.HypervisorConfig
 	q.pod = pod
 
-	if err = pod.storage.fetchHypervisorState(pod.id, &q.state); err != nil {
+	if err := pod.storage.fetchHypervisorState(pod.id, &q.state); err != nil {
 		q.Logger().Debug("Creating bridges")
 		q.state.Bridges = NewBridges(q.config.DefaultBridges, q.config.HypervisorMachineType)
+
+		q.Logger().Debug("Creating UUID")
+		q.state.UUID = uuid.Generate().String()
+
+		if err := pod.storage.storeHypervisorState(pod.id, q.state); err != nil {
```

Comment:
We should keep the hypervisor state storage either in pod.go , after the `hypervisor.init()` call.

Reply:
Well, this was done on purpose since I think it should be handled from the hypervisor implementation itself. Indeed, this state file is specific to every hypervisor implementation so that they can store some specific information. But let's say we have other hypervisor implementation that don't need to store/fetch anything, this means we would force them to store nothing.
I agree we would avoid code duplication if storing from pod.go, but in that case I think we should do everything from container/pod code. Or we go with everything controlled from the hypervisor implementation. The huge benefit with having this controlled from the hypervisor implementation is that we don't need to make assumption about when we should fetch/store from the pod/container.
I think we can take this change as is, and if you think we should move the entire fetch/store out of the qemu implementation because we can find a common pattern for all hypervisors about storage, then this should be done in a follow up PR.
Let me know what you think !

Reply:
Fair enough, I'm good with that.